modules/tfm/tfm/boards/common/attest_hal.c

```diff
@@ -15,34 +15,205 @@
 #include <nrf_cc3xx_platform.h>
 #include "tfm_strnlen.h"
 
-static enum tfm_security_lifecycle_t map_otp_lcs_to_tfm_slc(enum plat_otp_lcs_t lcs)
-{
+/* 16 is the only supported length */
+#define CONFIG_SB_PUBLIC_KEY_HASH_LEN 16
+
+/* implementation_id is a mandatory IAT claim */
+#define CONFIG_SB_IMPLEMENTATION_ID 1
+#include "bl_storage.h"
+#include <nrfx_nvmc.h>
+
+
+static enum tfm_security_lifecycle_t map_bl_storage_lcs_to_tfm_slc(enum lcs lcs){
 	switch (lcs) {
-	case PLAT_OTP_LCS_ASSEMBLY_AND_TEST:
+	case ASSEMBLY:
 		return TFM_SLC_ASSEMBLY_AND_TEST;
-	case PLAT_OTP_LCS_PSA_ROT_PROVISIONING:
+	case PROVISION:
 		return TFM_SLC_PSA_ROT_PROVISIONING;
-	case PLAT_OTP_LCS_SECURED:
+	case SECURE:
 		return TFM_SLC_SECURED;
-	case PLAT_OTP_LCS_DECOMMISSIONED:
+	case DECOMMISSIONED:
 		return TFM_SLC_DECOMMISSIONED;
-	case PLAT_OTP_LCS_UNKNOWN:
+	case UNKNOWN:
 	default:
 		return TFM_SLC_UNKNOWN;
 	}
 }
 
+static enum lcs map_tfm_slc_to_bl_storage_lcs(enum tfm_security_lifecycle_t lcs){
+	switch (lcs) {
+	case TFM_SLC_ASSEMBLY_AND_TEST:
+		return ASSEMBLY;
+	case TFM_SLC_PSA_ROT_PROVISIONING:
+		return PROVISION;
+	case TFM_SLC_SECURED:
+		return SECURE;
+	case TFM_SLC_DECOMMISSIONED:
+		return DECOMMISSIONED;
+	case TFM_SLC_UNKNOWN:
+	default:
+		return UNKNOWN;
+	}
+}
+
+/* This is temporary solution until the bl_storage API is available in TF-M
+ * This points to the UICR OTP region
+ */
+static const struct bl_storage_data *p_bl_storage_data =
+	(struct bl_storage_data *)PROVISION_PARTITION_ADDRESS;
+
+#define STATE_ENTERED 0x0000
+#define STATE_NOT_ENTERED 0xFFFF
+#define	EINVAL 22	/* Invalid argument */
+
+
+/** Function for reading a half-word (2 byte value) from OTP.
+ *
+ * \details The flash in OTP supports only aligned 4 byte read operations.
+ *          This function reads the encompassing 4 byte word, and returns the
+ *          requested half-word.
+ *
+ * \param[in]  ptr  Address to read.
+ *
+ * \return value
+ */
+static uint16_t read_halfword(const uint16_t *ptr)
+{
+	bool top_half = ((uint32_t)ptr % 4); /* Addr not div by 4 */
+	uintptr_t target_addr = (uintptr_t)ptr & ~3; /* Floor address */
+	uint32_t val32 = *(uint32_t *)target_addr;
+	__DSB(); /* Because of nRF9160 Erratum 7 */
+
+	return (top_half ? (val32 >> 16) : val32) & 0x0000FFFF;
+}
```

Comment:
TF-M has access to nrfx. So let's use it's functions:

/**
 * @brief Function for reading a 16-bit aligned halfword from the OTP (UICR)
 *
 * OTP is a region of the UICR present in some chips. This function must be used
 * to read halfword data from this region since unaligned accesses are not
 * available on the OTP flash area.
 *
 * @param address Address to read from. Must be halfword-aligned.
 *
 * @retval The contents at @p address.
 */
uint16_t nrfx_nvmc_otp_halfword_read(uint32_t address);

Reply:
god point, changed it